File: scripts/python/calculator/calculator.py
#!/usr/bin/env python3

import warnings
warnings.filterwarnings("ignore", category=DeprecationWarning)

import cgi

def main() -> None:
	form = cgi.FieldStorage()
	print("<html><body>")
	print("<h1>Python CGI Calculator</h1>")
	try:
		num1 = float(form.getvalue("num1", 0))
		num2 = float(form.getvalue("num2", 0))
		operation = form.getvalue("operation", "add")

		if operation == "add":
			result = num1 + num2
		elif operation == "sub":
			result = num1 - num2
		elif operation == "mul":
			result = num1 * num2
		elif operation == "div" and num2 != 0:
			result = num1 / num2
		else:
			result = "Invalid Operation"

		print(f"<h2>Result: {result}</h2>")

	except ValueError:
		print("<h2>Error: Invalid number</h2>")

	print("</body></html>")

if __name__ == "__main__":
	main()

# curl -X GET "http://localhost:8080/calculator?num1=10&num2=5&operation=div"

File: scripts/php/calculator/calculator.php
#!/usr/bin/env php

<?php
    $request_method = getenv("REQUEST_METHOD");

    echo "<html><body>";
    echo "<h1>PHP CGI Calculator</h1>";
    if ($request_method === 'GET') {
        parse_str(getenv("QUERY_STRING"), $query_params);

        try {
            $num1 = isset($query_params['num1']) ? floatval($query_params['num1']) : 0;
            $num2 = isset($query_params['num2']) ? floatval($query_params['num2']) : 0;
            $operation = isset($query_params['operation']) ? $query_params['operation'] : 'add';

            switch ($operation) {
                case 'add':
                    $result = $num1 + $num2;
                    break;
                case 'sub':
                    $result = $num1 - $num2;
                    break;
                case 'mul':
                    $result = $num1 * $num2;
                    break;
                case 'div':
                    if ($num2 != 0) {
                        $result = $num1 / $num2;
                    } else {
                        $result = 'Division by zero';
                    }
                    break;
                default:
                    $result = 'Invalid Operation';
                    break;
            }
            echo "<h2>Result: $result</h2>";

        } catch (Exception $e) {
            echo "<h2>Error: Invalid number</h2>";
        }
    } else {
        echo "<h2>Error: Invalid request method</h2>";
    }
    echo "</body></html>";

    // curl -X GET "http://localhost:8080/calculator?num1=10&num2=20&operation=add"
?>